library/ssl_tls.c

```diff
@@ -3142,6 +3142,29 @@ void mbedtls_ssl_init( mbedtls_ssl_context *ssl )
     memset( ssl, 0, sizeof( mbedtls_ssl_context ) );
 }
 
+static int ssl_conf_check(const mbedtls_ssl_context *ssl)
+{
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+
+    if( mbedtls_ssl_conf_is_tls13_only(ssl->conf) )
+    {
+        MBEDTLS_SSL_DEBUG_MSG( 4, ("ssl->conf is tls13 only."));
+        return( 0 );
+    }
+#endif
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
+    if( mbedtls_ssl_conf_is_tls12_only(ssl->conf) )
+    {
+        MBEDTLS_SSL_DEBUG_MSG( 4, ("ssl->conf is tls12 only."));
+        return( 0 );
+    }
+#endif
+    MBEDTLS_SSL_DEBUG_MSG( 4, ("ssl->conf is invalid."));
```

Comment:
Minor: I'd prefer something like `The SSL configuration is ...` instead of `ssl->conf`